Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -243,6 +244,20 @@ - (void)layerFailedToDecode:(NSNotification*)note
     });
 }
 
+- (void)layerRequiresFlushToResumeDecodingChanged:(NSNotification *)note
+{
+    RetainPtr<AVSampleBufferDisplayLayer> layer = (AVSampleBufferDisplayLayer *)[note object];
```

Comment:
no need for a dynamic cast or type assertion or anything?

Reply:
I mean, if the system delivers us a notification from an object we never signed up for notifications from, that would indicate much, much deeper problems. But sure, I'll ASSERT that the object is an AVSBDL.

Reply:
Actually, never mind, this should be handled by:
```
    if (!_layers.contains(layer.get()))
        return;
```
Better than a dynamic cast, this not only verifies that the object is an AVSBDL, it verifies that it is _this particular AVSBDL_.